src/main/java/com/idrsolutions/microservice/utils/FileDeletionService.java

```diff
@@ -0,0 +1,113 @@
+package com.idrsolutions.microservice.utils;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.Date;
+import java.util.List;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.stream.Stream;
+
+/**
+ * A class that will remove files after they exceed the individual's time to live (TTL).
+ * An example use would be to remove the input/output files for conversions once they exceed the TTL.
+ * When first run, it will immediately scan the directories and remove files that exceed the individual's TTL.
+ *
+ * If a new directory is used and requires a FileDeletionService, it is suggested to create a new instance.
+ */
+public class FileDeletionService {
+    private static final Logger LOG = Logger.getLogger(FileDeletionService.class.getName());
+
+    private final ScheduledExecutorService scheduledExecutorService = Executors.newScheduledThreadPool(1);
+
+    /**
+     * Create an instance of the FileDeletionService
+     *
+     * @param dirs the root directories that will be scanned for files that should be deleted
+     * @param fileLifeSpan the life span of the files before they should be deleted
+     * @param frequency the frequency in minutes that the service will run
+     */
+    public FileDeletionService(final String[] dirs, final long fileLifeSpan, final long frequency) {
+        setUpService(dirs, fileLifeSpan, frequency);
+    }
+
+    /**
+     * Use a ScheduledExecutorService to set up a service with the process of deleting any files that exceed the fileLifeSpan
+     *
+     * This method does not remove any previously set scheduled tasks from the ExecutorService
+     *
+     * @param dirs the root directories that will be scanned for files that should be deleted
+     * @param fileLifeSpan the life span of the files before they should be deleted
```

Comment:
Probably worth specifying the units for the lifespan too

Reply:
@JacobIDR Thank you for spotting. I have resolved in the latest commit